openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/DataContractRepository.java

```diff
@@ -208,6 +212,10 @@ protected void postUpdate(DataContract original, DataContract updated) {
     }
 
     postCreateOrUpdate(updated);
+    // Update test cases with dataContract reference for new/updated quality expectations
+    updateTestCasesWithDataContract(updated);
+    // Remove dataContract reference from test cases no longer in quality expectations
+    removeDataContractFromOldTestCases(original, updated);
```

Comment:
why do we update and remove?

Reply:
first step is to add the dataContract link to the new test cases in the DQ expectations
second is to remove it from those that have been removed from the DQ expectations